osu.Game.Rulesets.Osu/Skinning/SmokeSegment.cs

```diff
@@ -311,7 +314,11 @@ protected virtual Vector2 PointDirection(SmokePoint point)
                 return new Vector2(MathF.Sin(angle), -MathF.Cos(angle));
             }
 
-            private float nextRotation() => max_rotation * ((float)rotationRNG.NextDouble() * 2 - 1);
+            private float nextRotation()
+            {
+                rotationIndex++;
+                return max_rotation * (StatelessRNG.NextSingle(rotationIndex) * 2 - 1);
```

Comment:
Is there any particular reason why it's done this way? `NextSingle()` accepts a "seed" parameter and index into that seed.
So I'm proposing something like this:
```diff
diff --git a/osu.Game.Rulesets.Osu/Skinning/SmokeSegment.cs b/osu.Game.Rulesets.Osu/Skinning/SmokeSegment.cs
index 7af1b5d86a..aba4d0ff63 100644
--- a/osu.Game.Rulesets.Osu/Skinning/SmokeSegment.cs
+++ b/osu.Game.Rulesets.Osu/Skinning/SmokeSegment.cs
@@ -234,7 +234,7 @@ public sealed override void Draw(IRenderer renderer)
                 if (points.Count == 0)
                     return;

-                rotationIndex = rotationSeed;
+                rotationIndex = 0;

                 quadBatch ??= renderer.CreateQuadBatch<TexturedVertex2D>(max_point_count / 10, 10);
                 texture ??= renderer.WhitePixel;
@@ -314,11 +314,7 @@ protected virtual Vector2 PointDirection(SmokePoint point)
                 return new Vector2(MathF.Sin(angle), -MathF.Cos(angle));
             }

-            private float nextRotation()
-            {
-                rotationIndex++;
-                return max_rotation * (StatelessRNG.NextSingle(rotationIndex) * 2 - 1);
-            }
+            private float nextRotation() => max_rotation * (StatelessRNG.NextSingle(rotationSeed, rotationIndex++) * 2 - 1);

             private void drawPointQuad(SmokePoint point, RectangleF textureRect)

```

Reply:
Hm, seems like I misread the RNG code. For whatever reason I thought the series parameter was a bit shift, so after a certain index it just made everything 0. But no that's super dumb, it's the series that gets bit shifted by a constant amount. Your proposal is right.